third_party/amd/lib/TritonAMDGPUToLLVM/ElementwiseOpToLLVM.cpp

```diff
@@ -76,29 +87,67 @@ static Value cvtFp16ToFp32(Location loc, ConversionPatternRewriter &rewriter,
 }
 
 static Value cvtFp32ToFp16(Location loc, ConversionPatternRewriter &rewriter,
-                           const Value &v) {
+                           const Value &v, const RoundingMode rounding) {
   GCNBuilder builder;
+
   auto &cvt = *builder.create("v_cvt_f16_f32");
   auto res = builder.newOperand("=v");
   auto operand = builder.newOperand(v, "v");
+  if (rounding == RoundingMode::RTZ) {
+    auto &setRTZ = *builder.create("s_setreg_imm32_b32 0x1801, 0xc");
+    setRTZ();
+  }
   cvt(res, operand);
+  if (rounding == RoundingMode::RTZ) {
+    auto &resetRTZ = *builder.create("s_setreg_imm32_b32 0x1801, 0x0");
+    resetRTZ();
+  }
   return builder.launch(rewriter, loc, f16_ty, false);
 }
 
-static SmallVector<Value>
-convert_val_Fp16_to_Fp8(Location loc, ConversionPatternRewriter &rewriter,
-                        Value v0, Value v1, const std::string &fp8_format) {
+// convert fp8 to fp32
+static SmallVector<Value> cvtFp8ToFp32(Location loc,
+                                       ConversionPatternRewriter &rewriter,
+                                       Value v0, Value v1,
+                                       const std::string &fp8_format) {
   assert(fp8_format == "fp8" or fp8_format == "bf8");
-  std::string ins_str = "v_cvt_pk_" + fp8_format + "_f32";
+  std::string ins_str = "v_cvt_pk_f32_" + fp8_format;
 
-  auto f32_0 = cvtFp16ToFp32(loc, rewriter, v0);
-  auto f32_1 = cvtFp16ToFp32(loc, rewriter, v1);
+  auto fp8x4VecTy = vec_ty(i8_ty, 4);
+  Value fp8x4Vec = undef(fp8x4VecTy);
+  fp8x4Vec = insert_element(fp8x4VecTy, fp8x4Vec, v0, i32_val(0));
+  fp8x4Vec = insert_element(fp8x4VecTy, fp8x4Vec, v1, i32_val(1));
+  auto i32v = bitcast(fp8x4Vec, i32_ty);
+
+  GCNBuilder builder1;
+  auto &cvt = *builder1.create(ins_str);
+  auto res = builder1.newOperand("=v");
+  auto operand = builder1.newOperand(i32v, "v");
+  cvt(res, operand);
+  auto i64v = builder1.launch(rewriter, loc, i64_ty, false);
+  auto fp32x2VecTy = vec_ty(f32_ty, 2);
+  auto fp32x2Vec = bitcast(i64v, fp32x2VecTy);
+
+  SmallVector<Value> ret(2);
+  ret[0] = extract_element(f32_ty, fp32x2Vec, i32_val(0));
+  ret[1] = extract_element(f32_ty, fp32x2Vec, i32_val(1));
+
+  return ret;
+}
+
+// convert fp32 to fp8
+static SmallVector<Value> cvtFp32ToFp8(Location loc,
+                                       ConversionPatternRewriter &rewriter,
+                                       Value v0, Value v1,
+                                       const std::string &fp8_format) {
+  assert(fp8_format == "fp8" or fp8_format == "bf8");
```

Comment:
Similarly here and other places we `assert`.

Reply:
Done.